### PR TITLE
Add CI for frontend build and lint

### LINKED ISSUE #60: Add a Github workflow that builds the frontend
I was trying to deploy the frontend to heroku and I came across lots of type errors when Heroku ran `yarn build`.  These errors should have been caught by the CI before the respective PRs were merged.

### PR DESCRIPTION
**Pull request by Dhaiwat10**
Closes #60

### CODE REVIEW

**Comment by damianmarti** [comment]
@Dhaiwat10 LGTM, just add the needed env vars on .github/workflows/ci.yaml

https://docs.github.com/en/actions/learn-github-actions/environment-variables

**Comment by Dhaiwat10** [comment]
@damianmarti I saw that, my only worry is that this article is aimed at repo secrets rather than `.env` variables that our codebase can read. Nevertheless, I'll give this a go and see if it works

**Comment by damianmarti** [comment]
> @damianmarti I saw that, my only worry is that this article is aimed at repo secrets rather than `.env` variables that our codebase can read. Nevertheless, I'll give this a go and see if it works

https://damienaicheh.github.io/github/actions/2021/04/15/environment-variables-secrets-github-actions-en

A post explaining how to set env vars from github secrets.

**Comment by Dhaiwat10** [comment]
Lol @damianmarti I just pushed a fix and you shared it. Coincidence

Anyways I think it's working now :)

**Comment by Dhaiwat10** [comment]
I used this comment: https://stackoverflow.com/a/69082660

**Comment by Dhaiwat10** [comment]
### Note:

I have disabled the linting step temporarily because it was throwing some weird error related to the eslint config. I'm opening a separate issue for that.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+env:
+  NEXT_PUBLIC_INFURA_ID: ${{secrets.NEXT_PUBLIC_INFURA_ID}}
+  WEB3STORAGE_TOKEN: ${{secrets.WEB3STORAGE_TOKEN}}
```
Comment (Dhaiwat10) [inline .github/workflows/ci.yaml:13]: Are there any more `.env` variables that we need? @damianmarti

**Pull request merged**